src/run/run.c

```diff
@@ -1660,12 +1662,44 @@ static int run_context_reconnect(RunContext *c) {
         return run_context_update(c);
 }
 
+static int run_context_check_started(RunContext *c, const char *current_job) {
+        int r;
+
+        assert(c);
+
+        if (!c->start_job)
+                return 0; /* Already started? */
+
+        if (streq_ptr(c->start_job, current_job))
+                return 0; /* The start job is still active. */
+
+        /* The start job is finished. */
+        c->start_job = mfree(c->start_job);
+        c->match_job_removed = sd_bus_slot_unref(c->match_job_removed);
+
+        /* Setup ptyfwd now if --pty-late is specified. */
+        r = run_context_setup_ptyfwd(c);
+        if (r < 0)
+                return sd_event_exit(c->event, r);
```

Comment:
Not your fault, but we probably should not use r as exit status. Exit with EXIT_FAILURE and propagate r?